Sources/BraintreeShopperInsights/BTShopperInsightsClient.swift

```diff
@@ -31,8 +31,12 @@ public class BTShopperInsightsClient {
     /// - Returns: A `BTShopperInsightsResult` instance
     /// - Note: This feature is in beta. It's public API may change or be removed in future releases.
     public func getRecommendedPaymentMethods(request: BTShopperInsightsRequest) async throws -> BTShopperInsightsResult {
+        apiClient.sendAnalyticsEvent(BTShopperInsightsAnalytics.recommendedPaymentsStarted)
+        
         if isVenmoAppInstalled() && isPayPalAppInstalled() {
-            return BTShopperInsightsResult(isPayPalRecommended: true, isVenmoRecommended: true)
+            let result = BTShopperInsightsResult(isPayPalRecommended: true, isVenmoRecommended: true)
+            notifySuccess(with: result)
+            return result
```

Comment:
For folks reviewing, the other pattern Alekhya & I discussed looked something like:

```
// Callsite
return notifySuccess(with: result)

// Method definition
func notifySuccess(with result: BTShopperInsightsResult) -> BTShopperInsightsResult {
    apiClient.sendAnalyticEvent(...)
    return result
}
```

Reply:
Since we're using an async-await method here, we can't follow the exact pattern we've established for analytic helper methods which pass a completion around (see [here](https://github.com/braintree/braintree_ios/blob/f11dccb5b9f9f7ee67a8de8f60262ff290a4604d/Sources/BraintreeCard/BTCardClient.swift#L206-L220)).

What option do folks find more clear?

Reply:
I like the above proposed in this comment thread.

Reply:
I assume we can make a similar thow-ing method for notify failure as well, but yeah I like what you proposed Sammy and think it's most similar to what we do today.

Reply:
thanks for the suggestions y'all, addressed in 5c16b60b274c998158563c2f6a4080e2aa991891